Demo page type and templates

mgax opened the pull request:
Fix https://github.com/wagtail/wagtail-newsletter/issues/5.

- ~To spin up the site locally, simply run `tox -e interactive`, it's the same site that is part of the testsuite.~ The demo project is now in the `/demo` subdirectory.
- The code includes our StreamField migration workaround, to keep the block schema from generating unnecessary migrations.
- I've opted for a simple content schema: one page type with the fields `author`, `date`, and a StreamField `body`.
- The body currently defines block types for rich text, image, and raw HTML, because they seem particularly relevant to newsletter content rendering. I wonder if it makes sense to add anything else at this point. We can easily add more later without generating migration noise.

thibaudcolas reviewed (commented):
👍 code is looking good. One change I’d recommend is move the demo site from `wagtail_newsletter/test` to a top-level `demo` folder. Specifically so we:

- Keep unit tests and demo site code clearly separate
- Avoid shipping demo code in the PyPI package

mgax commented:
@thibaudcolas thanks! I've copied the demo site to `/demo` and restored the previous `test` site. I've also simplified the configuration and added a note in the README.

thibaudcolas reviewed (approved):
APPROVED
